tests/settings_test.py

```diff
@@ -125,3 +127,23 @@ def test_database():
             mock_subprocess.return_value = ""
             with pytest.raises(DatabaseSourceException):
                 assert settings.database
+
+
+def test_mask_passsword_in_logs():
+    # type: () -> None
+    settings = Settings()
+    settings.database_source = "/path/to/program"
+    test_url = "mysql://user:password@example.com:8888/merou"
+    expected_url = "mysql://user:???@example.com:8888/merou"
+
+    log_output = StringIO()
+    log_handler = logging.StreamHandler(log_output)
+    settings._logger.addHandler(log_handler)
+    settings._logger.setLevel(logging.DEBUG)
+
+    with patch("subprocess.check_output") as mock_subprocess:
+        mock_subprocess.return_value = "mysql://user:password@example.com:8888/merou"
+        assert settings.database == test_url
```

Comment:
Nit: a comment calling out that this is where the logging actually happens would be nice :)

Reply:
Yeah, good idea.